### PR TITLE
Grafana alloy update

### PR DESCRIPTION
**Pull request by JonahEdick**
Fixes:

Related:

#### For new package PRs only

- [ ] This PR is marked as fixing a pre-existing package request bug
  - [ ] Alternatively, the PR is marked as related to a pre-existing package request bug, such as a dependency
- [x] REQUIRED - The package is available under an OSI-approved or FSF-approved license
- [x] REQUIRED - The version of the package is still receiving security updates
- [ ] This PR links to the upstream project's support policy (e.g. `endoflife.date`)

#### For new version streams

- [ ] The upstream project actually supports multiple concurrent versions.
- [ ] Any subpackages include the version string in their package name (e.g. `name: ${{package.name}}-compat`)
- [ ] The package (and subpackages) `provides:` logical unversioned forms of the package (e.g. `nodejs`, `nodejs-lts`)
- [ ] If non-streamed package names no longer built, open PR to withdraw them (see [WITHDRAWING PACKAGES](https://github.com/wolfi-dev/os/blob/main/WITHDRAWING_PACKAGES.md))

#### For package updates (renames) in the base images

When updating packages part of base images (i.e. cgr.dev/chainguard/wolfi-base or ghcr.io/wolfi-dev/sdk)
- [ ] REQUIRED cgr.dev/chainguard/wolfi-base and ghcr.io/wolfi-dev/sdk images successfully build
- [ ] REQUIRED cgr.dev/chainguard/wolfi-base and ghcr.io/wolfi-dev/sdk contain no obsolete (no longer built) packages
- [ ] Upon launch, does `apk upgrade --latest` successfully upgrades packages or performs no actions

#### For security-related PRs

- [ ] The security fix is recorded in the [advisories](https://github.com/wolfi-dev/advisories) repo

#### For version bump PRs

- [ ] The `epoch` field is reset to 0

#### For PRs that add patches

- [ ] Patch source is documented

### CODE REVIEW

File: grafana-alloy.yaml [hunk 1]
```diff
@@ -12,6 +12,9 @@ environment:
       - busybox
```
Comment (developer-guy) [inline grafana-alloy.yaml:12]: you need to bump the epoch

File: grafana-alloy.yaml [hunk 2]
```diff
@@ -26,6 +29,9 @@ pipeline:
     with:
       deps: github.com/hashicorp/go-retryablehttp@v0.7.7 google.golang.org/grpc@v1.64.1
 
+  - name: Generate UI
+    runs: make generate-ui
```
Comment (jamonation) [inline grafana-alloy.yaml:33]: Where do these files end up @JonahEdick? I don't see anything in the final packages.

The build log shows this:

```
2024/07/30 19:21:26 INFO Compiled successfully.
2024/07/30 19:21:26 INFO
2024/07/30 19:21:26 INFO File sizes after gzip:
2024/07/30 19:21:26 INFO
2024/07/30 19:21:26 INFO   409.57 kB  build/static/js/main.cba49e5e.js
2024/07/30 19:21:26 INFO   4.25 kB    build/static/css/main.314ea319.css
2024/07/30 19:21:26 INFO
2024/07/30 19:21:26 INFO The project was built assuming it is hosted at ./public/.
2024/07/30 19:21:26 INFO You can control this with the homepage field in your package.json.
2024/07/30 19:21:26 INFO
2024/07/30 19:21:26 INFO The build folder is ready to be deployed.
```

But when I install the package I don't see any of those files. Are they compiled into the binary, or should we have a subpackage for the UI?

Comment (JonahEdick) [reply grafana-alloy.yaml:33]: According to Grafana's docker file, it states the following
```
# Build the UI before building Alloy, which will then bake the final UI into
# the binary.
```
So by that logic I would assume that there isn't any extra files because they get put into the same grafana-alloy binary that we were building before

Comment (developer-guy) [reply grafana-alloy.yaml:33]: https://github.com/grafana/alloy/blob/f439621443311d1c293f16162f35083c8d45cbc3/internal/web/ui/assets_builtin.go#L15 here!

Comment (jamonation) [reply grafana-alloy.yaml:33]: When I tried installing our package and running the UI I get a 404 page. But running the image from their Docs shows it at 127.0.0.1:12345

https://grafana.com/docs/alloy/latest/set-up/install/docker/

So I think we need to figure out how to get the yarn/npm assets into the right place for go build to pick them up.

**Comment by developer-guy** [comment]
What about these following two items we noted?

- They copy the different config.alloy file than ours, [here](https://github.com/grafana/alloy/blob/c9663c58cc6ec58d97f33ecf783465678e553ce6/Dockerfile#L48), we copied packaging/config.alloy into our package but they copied example-config.alloy which is quite different than ours.
- They specified a bunch of runtime dependencies which we don't specify in our package:

**Comment by JonahEdick** [comment]
> What about these following two items we noted?
> 
> * They copy the different config.alloy file than ours, [here](https://github.com/grafana/alloy/blob/c9663c58cc6ec58d97f33ecf783465678e553ce6/Dockerfile#L48), we copied packaging/config.alloy into our package but they copied example-config.alloy which is quite different than ours.
> * They specified a bunch of runtime dependencies which we don't specify in our package:

I have added the packages mentioned in the docker file and in regards to the config.alloy being different, I don't believe that should have a major impact on the final outcome since they both do come from the official source but I have been testing with the example-config.alloy file and hasn't yielded any different results yet. Is there another place that would mention more dependencies?

File: grafana-alloy.yaml (hunk 2 above)
Comment (developer-guy) [reply grafana-alloy.yaml:36]: Found: https://github.com/grafana/alloy/blob/b068c13bab9ff8f6351c68096a9e18b58acda1eb/Dockerfile#L29C4-L29C61

TLDR;

_GO_TAGS="netgo builtinassets promtail_journal_enabled"_

File: grafana-alloy.yaml [hunk 3]
```diff
@@ -22,10 +25,16 @@ pipeline:
       tag: v${{package.version}}
       expected-commit: e83e8af406b19c11b102b76f21142f7561a77036
 
+  - name: Generate UI
```
[the remark this answers is not in the document]
Comment (JonahEdick) [reply grafana-alloy.yaml:28]: This was a suggestion Jamon made on trying to put this task before we go/bump and I forgot to remove it below and I must have forgot to remove it.

File: grafana-alloy.yaml (hunk 2 above)
Comment (JonahEdick) [reply grafana-alloy.yaml:36]: How would one go at adding tags to the go/build process on the pipeline?

Comment (developer-guy) [reply grafana-alloy.yaml:36]: https://github.com/chainguard-dev/melange/blob/5beea3ebd2162328f01d756674091797e8af88d1/pkg/build/pipelines/go/build.yaml#L22

Comment (JonahEdick) [reply grafana-alloy.yaml:36]: good news! 
```
/work/packages # wget localhost:12345
--2024-07-31 19:17:49--  http://localhost:12345/
Resolving localhost... ::1, 127.0.0.1
Connecting to localhost|::1|:12345... failed: Connection refused.
Connecting to localhost|127.0.0.1|:12345... connected.
HTTP request sent, awaiting response... 200 OK
Length: 707 [text/html]
index.html: Read-only file system
```
I was able to retrieve the webpage on the local-wolfi container running grafana-alloy! (I can verify that it is the expected outcome after making an image with it)

Comment (developer-guy) [reply grafana-alloy.yaml:33]: I made some changes on the PR could you test it if that works as you expected? 

Also tried to bump the docker deps but the compilation failed with:

```
2024/07/31 19:44:08 INFO Done in 39.15s.
2024/07/31 19:44:08 INFO running step "Run a build using the go compiler"
2024/07/31 19:44:55 WARN # github.com/google/cadvisor/container/docker/utils
2024/07/31 19:44:55 WARN /var/cache/melange/gomodcache/github.com/grafana/cadvisor@v0.0.0-20231110094609-5f7917925dea/container/docker/utils/docker.go:48:42: undefined: dockertypes.Info
2024/07/31 19:44:55 WARN /var/cache/melange/gomodcache/github.com/grafana/cadvisor@v0.0.0-20231110094609-5f7917925dea/container/docker/utils/docker.go:57:44: undefined: dockertypes.Info
2024/07/31 19:44:55 WARN /var/cache/melange/gomodcache/github.com/grafana/cadvisor@v0.0.0-20231110094609-5f7917925dea/container/docker/utils/docker.go:77:43: undefined: dockertypes.Info
2024/07/31 19:44:55 WARN /var/cache/melange/gomodcache/github.com/grafana/cadvisor@v0.0.0-20231110094609-5f7917925dea/container/docker/utils/docker.go:86:48: undefined: dockertypes.ImageSummary
```

which is good because we could use that information while filling the note of the advisory data.

**Comment by developer-guy** [comment]
I mean by dependencies is this:

- https://github.com/grafana/alloy/blob/a014bda1725d7817d34523f956812d3ae5e0c809/Dockerfile#L41-L44

File: grafana-alloy.yaml (hunk 2 above)
Comment (JonahEdick) [reply grafana-alloy.yaml:33]: Yeah I can pull and test this. One second

Comment (jamonation) [reply grafana-alloy.yaml:33]: Looks good now, I can see the UI!

File: grafana-alloy.yaml [hunk 4]
```diff
@@ -12,6 +12,9 @@ environment:
       - busybox
       - ca-certificates-bundle
       - go
+      - nodejs
```
Comment (tuananh) [inline grafana-alloy.yaml:15]: should we pin nodejs to a major version?

Comment (developer-guy) [reply grafana-alloy.yaml:15]: they seem using nodejs-18: https://github.com/grafana/alloy/blob/a014bda1725d7817d34523f956812d3ae5e0c809/tools/build-image/Dockerfile#L71-L72

Comment (developer-guy) [reply grafana-alloy.yaml:15]:
```
Status: Downloaded newer image for grafana/alloy-build-image:v0.1.3
root@0ef59700d4da:/# node --version
v18.20.4
root@0ef59700d4da:/#
```

lets use nodejs-18

**Comment by developer-guy** [comment]
according to their makefile https://github.com/grafana/alloy/blob/f5043307c5b9bb6d94dbb0d3c4d2bc19139f4b63/Makefile#L89 we have to enable cgo so I have enabled it and added `systemd-dev` which is required to build the project.

**Review by jamonation — APPROVED** [review]
